srcts/src/components/sidebar.ts

```diff
@@ -28,20 +63,75 @@ class Sidebar {
     TRANSITIONING: "transitioning",
   };
 
-  public static initCollapsible(container: HTMLElement): void {
-    // Signal that this layout is initialized by removing the init attribute
-    container.removeAttribute("data-bslib-sidebar-init");
+  private static onReadyScheduled = false;
+  private static instanceMap: WeakMap<HTMLElement, Sidebar> = new WeakMap();
+
+  public static getInstance(el: HTMLElement): Sidebar | undefined {
+    return Sidebar.instanceMap.get(el);
+  }
+
+  public static initCollapsibleAll(scope: Document | HTMLElement): void {
+    if (document.readyState === "loading") {
+      if (!Sidebar.onReadyScheduled) {
+        Sidebar.onReadyScheduled = true;
+        document.addEventListener("DOMContentLoaded", () =>
+          Sidebar.initCollapsibleAll(document)
+        );
+      }
+      return;
+    }
 
-    Sidebar._initSidebarCounters(container);
-    Sidebar._initDesktop(container);
+    const initSelector = "[data-bslib-sidebar-init]";
+    let scopeMatches = false;
+    if (scope instanceof HTMLElement) {
+      scopeMatches = scope.matches(initSelector);
+    }
+    if (!(scopeMatches || scope.querySelector(initSelector))) {
+      // no sidebars to initialize
+      return;
+    }
+
+    const containers = Array.from(scope.querySelectorAll(initSelector));
+    if (scopeMatches && scope instanceof HTMLElement) {
+      containers.unshift(scope);
+    }
+
+    containers.forEach((container) => new Sidebar(container as HTMLElement));
```

Comment:
If we have no reason to support anything but `document`, let's keep it simple

Reply:
Done in 45587166a22992655c88205eb8fa2b2fb624f1e2